src/components/avatar.rs

```diff
@@ -194,29 +250,23 @@ impl Avatar {
     pub(crate) fn set_item(&self, item: Option<glib::Object>) {
```

Comment:
The function should rather look like
```rust
    pub(crate) fn set_item(&self, item: Option<glib::Object>) {
        let imp = self.imp();

        imp.chat_signal_group
            .get()
            .unwrap()
            .set_target(item.as_ref().and_then(|item| item.downcast_ref::<Chat>()));
        imp.user_signal_group
            .get()
            .unwrap()
            .set_target(item.as_ref().and_then(|item| item.downcast_ref::<User>()));

        imp.item.replace(item);

        self.update_display_name();
        self.update_avatar();

        self.notify("item");
    }
```
- Much easier to read
- Sets one of the signal groups to `None`. Otherwise a chat might still be updating the avator after it has been replaced by a user.